### ISSUE TITLE
Is it possible to play sound from memory? [question]

### ISSUE THREAD

**Issue by bockoblur** [luanpotter/audioplayers#182]
I have short sound clips (~1-5s) that I generate at runtime.
They are stored in `List<int>` or `Int16List` type variable and correspond to simple wav format.
Is it somehow possible to pass this variable/buffer instead of filename parameter to ` .play() ` or `.loop()` methods, so the player will play (or loop) it from memory, without writing the file to temporary folder?

I am now using:
`myPlayer.play( Uri.dataFromBytes( myList, mimeType: "audio/wav"), isLocal:false )`
to pass `dataUri` instead of file name and it works, but that doesn't seem too efficient as passing the memory buffer directly to play method.

**Comment by erickzanardo** [comment]
Are you facing any problems with this code?

It seems ok to me. Writing a temp file could be a good solution too, so you can cache the files.

**Comment by bockoblur** [comment]
Code works fine, but it is not very efficient. It takes more time to make dataUri than it takes to generate sample, and I was just wondering is there a possibility to play a sound directly from memory, since it's already there?
I am not Android nor iOS expert, maybe it's not even possible, but I was hoping to find out. Thanks.

**Comment by erickzanardo** [comment]
I see, right now, there isn't a direct away to play from the memory, this is something that we want to do eventually, but still didn't had time to see how that would be achieved, especially on iOs.

One workaround for you could be writing a temporary file, temp files are written on the memory, so it should be way faster than the uri generation.

I will keep this open to remind us about this enhancement. Thanks

**Comment by bardiarastin** [comment]
hi @erickzanardo, thanks for awesome project.
what i'm trying to build is a music app that when you  press play button the song gets both played and cached.

i'm looking for a way to download a mp3 file from server and do both playing and writing it to a file on user device simultaneously. 

you said that "you could be writing a temporary file" but I wanna keep that file forever on user device, any idea or samples maybe?!

one more question that I have is that I don't know how to handle loading state, there is nothing anything such as isLoading or isBuffering in AudioPlayerState :(

one more thing is that there is no queue concept so I can give it a list of songs with next and previous methods, I guess I have to write a wrapper for this.

**Comment by DeRealMorgan** [comment]
If I am not reading that wrong my question was a different ome.

Right now it is not possible to create calling apps because you cannot replay the caputed data right away (after sending it to the other user). There are some 3rd party libs but none work uniformly for all platforms or at least for android/ios.

**Comment by Aoi-hosizora** [comment]
Oh I really need this feature, I need to play a base64 encoding ogg.

**Comment by bockoblur** [comment]
> Oh I really need this feature, I need to play a base64 encoding ogg.

I just found [soundpool](https://github.com/ukasz123/soundpool). Haven't tried it yet, but it looks like it can solve this issue. Plays short sounds from `ByteData`.

**Comment by Aoi-hosizora** [comment]
@bockoblur Thanks, I will try that later.

**Comment by UttamPanchasara** [comment]
Any update on this?

**Comment by erickzanardo** [comment]
`playBytes` is available now.

**Comment by yadames** [comment]
> `playBytes` ya está disponible.

In which version is it fixed? I am using version 0.18.3 and in iOS it does not appear available. Basically there is an "if" that throws an exception on any platform other than Android

![image](https://user-images.githubusercontent.com/47356433/116460834-826f5700-a835-11eb-9bc1-84230711bd37.png)

**Comment by S-Man42** [comment]
The TO previously used:
````
myPlayer.play( Uri.dataFromBytes( myList, mimeType: "audio/wav"), isLocal:false )
````
Does this also work for iOS? What about web support?
